### PR TITLE
mariadb: 10.2.16 -> 10.2.17

### PR DESCRIPTION
**Pull request by andir**
###### Motivation for this change

mariadb: 10.2.16 -> 10.2.17

Bump to latest stable version of the 10.2.x branch. Besides many bug fixes the
following security related issues have been fixed:
 - CVE-2018-3060
 - CVE-2018-3064
 - CVE-2018-3063
 - CVE-2018-3058
 - CVE-2018-3066

Also includes a test for mariadb based on the existing mysql test.

This should be considered for backporting since the version in **18.03** has even more open issues...

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [X] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by andir** [comment]
@GrahamcOfBorg test mysql

File: nixos/tests/mysql.nix [hunk 1]
```diff
@@ -1,24 +1,34 @@
...
-      };
-  };
+    nodes = {
+      master =
+        { pkgs, ... }:
+
+        {
+          services.mysql.enable = true;
+          services.mysql.initialDatabases = [ { name = "testdb"; schema = ./testdb.sql; } ];
+          services.mysql.package = pkgs.mysql;
```
Comment (vcunat) [inline nixos/tests/mysql.nix]: Are you sure this changes for the different tests?

**Comment by vcunat** [comment]
Staged for 18.03 as 8a804f70bc.

File: nixos/tests/mysql.nix (hunk 1 above)
Comment (andir) [reply nixos/tests/mysql.nix]: Ups, I changed it :)

File: nixos/tests/mysql.nix [hunk 2]
```diff
@@ -1,24 +1,34 @@
-import ./make-test.nix ({ pkgs, ...} : {
-  name = "mysql";
-  meta = with pkgs.stdenv.lib.maintainers; {
-    maintainers = [ eelco chaoflow shlevy ];
+{ system ? builtins.currentSystem }:
+with import ../lib/testing.nix { inherit system; };
+let
+  packages = {
+    mysql = pkgs.mysql;
```
Comment (vcunat) [inline nixos/tests/mysql.nix]: I suppose you don't know:
```
nix-repl> mysql.name
"mariadb-10.2.16"
```

**Comment by xeji** [comment]
@andir these "2" tests are the same since `mysql` is an alias for `mariadb`. If you really want mysql, use `pkgs.mysql57`.

**Comment by andir** [comment]
I got that from vcunat's comment. Just didn't have the time to fix it yet.

**Comment by xeji** [comment]
Great, sorry for the noise.

**Comment by luzpaz** [comment]
Should probably be closed in favor of https://github.com/NixOS/nixpkgs/pull/44343

**Comment by vcunat** [comment]
This change had been pushed as 5da1d01bf2d.

**Pull request closed without merging**